### DISCUSSION TITLE
Global background-image settings

### DISCUSSION THREAD

**Discussion by agerlach** [quarto-dev/quarto-cli#1307] [Q&A] [answered]
Howdy. I would like to set a `background-image` with `background-size: contain` for all slides in a RevealJS presentation. Is this currently supported?

Setting

```
format:
  revealjs:
    background-image: "assets/myimage.png"
    background-size: contain 
...
```

includes the image as a background but it is much larger than the slide and appears to be `background-size=100%`

Thanks

**Comment by mcanouil** [comment]
It should work.
At least as per the example in https://quarto.org/docs/presentations/revealjs/index.html#image-backgrounds

The default for `background-size` is `cover`.

**Comment by agerlach** [comment]
It does work as shown when specified for an individual slide but does not
work when specified in the format header as shown in op.

**Comment by cderv** [comment, marked as answer]
I believe the only global option for revealjs are the `parallax-*` ones: https://quarto.org/docs/reference/formats/presentations/revealjs.html#media

````yaml
format:
  revealjs:
    background-image: "assets/myimage.png"
    parallax-background-size: contain 
````

`background-size` is not among [global option you can set for revealjs](https://quarto.org/docs/reference/formats/presentations/revealjs.html) and `background-image` as a global option is just an alias for `parallax-background-image` that [Pandoc knows how to handle](https://pandoc.org/MANUAL.html#on-all-slides-beamer-reveal.js-pptx) because `background-image` is also used on other slide format (beamer or pptx) - (see [in template](https://github.com/jgm/pandoc/blob/7fdc01ac0d91a037b48847939b762654174b4125/data/templates/default.revealjs#L235-L244)

If parallax is not what you need, you can always use a Lua filter to to apply some classes and attributes to all element of your content, like the slide header. 

````lua
function Header (el)
  -- I am using level 2 header for slides in the markdown below
  if el.level == 2 then
    el.attributes['background-image'] = 'https://revealjs.com/images/logo/reveal-black-text.svg'
    el.attributes['background-size'] = 'contain'
  end
  return el
end
````

`````markdown
---
title: "Untitled"
format: 
  revealjs: 
    filters: 
      - filter.lua
---

## Slide 1

Test

## Slide 2

Test
`````

Hope it helps

**Reply by agerlach** [reply]
Thanks for teaching a man to fish 🎣
